transport/api/v1/newsletter.go

```diff
@@ -49,3 +51,145 @@ func (h *Handler) CreateNewsletter(w http.ResponseWriter, r *http.Request) {
 		Description: description,
 	})
 }
+
+func (h *Handler) DeleteNewsletter(w http.ResponseWriter, r *http.Request) {
+	idUUID, err := uuid.Parse(chi.URLParam(r, "newsletterId"))
+	editorId := r.Context().Value("editor_id")
+	editorIdUUID, err := uuid.Parse(editorId.(string))
+	// validation of newsletter ID
+	if err != nil {
+		util.WriteErrResponse(w, http.StatusBadRequest, errors.New("invalid newsletter ID"))
+		return
+	}
+	// validation of editor ID
+	if err != nil {
```

Comment:
Tady je to přeházený. na řádku 56 máš err a hned na řádku 58 ho přepíšeš errorem z druhého uuid.Parse bez kontroly err != nil (to je až na řádku 60 a 64 - což v obou případech je error z druhého parsu.